osquery/tables/applications/posix/prometheus_metrics.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+#include <sstream>
+
+#include <osquery/config.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include <osquery/core/conversions.h>
+
+#include <osquery/tables/applications/posix/prometheus_metrics.h>
+#include <osquery/tables/applications/posix/prometheus_metrics_utils.h>
+
+namespace http = boost::network::http;
+
+namespace osquery {
+namespace tables {
+
+void parseScrapeResults(const std::map<std::string, retData>& scrapeResults,
+                        QueryData& rows) {
+  for (auto const& target : scrapeResults) {
+    std::stringstream ss(target.second.content);
+    std::string dest;
+    std::string ts(std::to_string(target.second.timestampMS.count()));
+
+    while (std::getline(ss, dest)) {
+      if (!dest.empty() && dest[0] != '#') {
+        std::vector<std::string> metric(osquery::split(dest, " "));
+
+        if (metric.size() > 1) {
+          Row r;
+          r[kColTargetName] = target.first;
+          r[kColTimeStamp] = ts;
+          r[kColMetric] = metric[0];
+          r[kColValue] = metric[1];
+
+          rows.push_back(r);
+        }
+      }
+    }
+  }
+}
+
+void scrapeTargets(std::map<std::string, retData>& scrapeResults) {
+  http::client client(
+      http::client::options().follow_redirects(true).timeout(1));
+
+  for (auto& target : scrapeResults) {
+    try {
+      http::client::request request(target.first);
+      http::client::response response(client.get(request));
+
+      target.second.timestampMS =
```

Comment:
Out of curiosity, why is this in milliseconds? Generally timestamps are in seconds and this will give it more consistency with other tables, is there a reason you want the extra precision?